So this was originally applied in #2695:
> When compiling on FreeBSD, the calculation here returns an unsigned int. This causes a compile-time error with std::min, which cannot compare signed with unsigned integers.
This pull inserts an explicit cast, treating the calculated value as signed, keeping the compiler happy.

Then reverted in #9598 to:
> Improve readability by removing redundant casts to same type (on all platforms)

Now we're re-adding (int) again but only for FreeBSD.

I've just built master (3f398d7) successfully on a FreeBSD 11.1 VM, so I'm curious to know which version of FreeBSD is this broken on?

Regardless of the above, the current patch doesn't compile:
```
fvisibility=hidden -c -o libbitcoin_server_a-init.o `test -f 'init.cpp' || echo './'`init.cpp
init.cpp:969:2: error: invalid preprocessing directive #fi
 #fi
  ^~
init.cpp:964:0: error: unterminated #else
 #if defined(__FreeBSD__) || defined(__DragonFly__)
 
make[2]: *** [libbitcoin_server_a-init.o] Error 1
Makefile:5817: recipe for target 'libbitcoin_server_a-init.o' failed
```

--- a/src/init.cpp
+++ b/src/init.cpp
@@ -961,7 +961,8 @@ bool AppInitParameterInteraction()
     nMaxConnections = std::max(nUserMaxConnections, 0);
 
     // Trim requested connection counts, to fit into system limitations
-    nMaxConnections = std::max(std::min(nMaxConnections, FD_SETSIZE - nBind - MIN_CORE_FILEDESCRIPTORS - MAX_ADDNODE_CONNECTIONS), 0);
+    // <int> in std::min<int>(...) to work around FreeBSD compilation issue described in #2695
+    nMaxConnections = std::max(std::min<int>(nMaxConnections, FD_SETSIZE - nBind - MIN_CORE_FILEDESCRIPTORS - MAX_ADDNODE_CONNECTIONS), 0);
     nFD = RaiseFileDescriptorLimit(nMaxConnections + MIN_CORE_FILEDESCRIPTORS + MAX_ADDNODE_CONNECTIONS);
     if (nFD < MIN_CORE_FILEDESCRIPTORS)
         return InitError(_("Not enough file descriptors available."));

